### PR TITLE
Added new files in KB

### LINKED ISSUE #176: [KB] Add KB for docker/metadata-action
harden-runner-link: 
https://app.stepsecurity.io/github/microsoft/msquic/actions/runs/1723947678

### LINKED ISSUE #179: [KB] Add KB for fossas/fossa-action
harden-runner-link: 
https://app.stepsecurity.io/github/MTRNord/matrix-art/actions/runs/1724838465

### LINKED ISSUE #181: [KB] Add KB for wei/curl
harden-runner-link: 
https://app.stepsecurity.io/github/MTRNord/matrix-art/actions/runs/1725090093

### LINKED ISSUE #190: [KB] Add KB for ossf/scorecard-action
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/bl3auto/actions/runs/1731157027

### LINKED ISSUE #192: [KB] Add KB for github/super-linter/slim
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/ghafiles/actions/runs/1731635057

### LINKED ISSUE #194: [KB] Add KB for anchore/sbom-action/download-syft
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/bl3auto/actions/runs/1732141963

### LINKED ISSUE #195: [KB] Add KB for zricethezav/gitleaks-action
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/ghafiles/actions/runs/1732252131

### LINKED ISSUE #197: [KB] Add KB for golangci/golangci-lint-action
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/bl3auto/actions/runs/1732342395

### LINKED ISSUE #198: [KB] Add KB for snyk/actions/golang
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/bl3auto/actions/runs/1732366946

### LINKED ISSUE #199: [KB] Add KB for securego/gosec
harden-runner-link: 
https://app.stepsecurity.io/github/jauderho/bl3auto/actions/runs/1732366946

### PR DESCRIPTION
**Pull request by h0x0er**
Fixes #176 
Fixes #179
Fixes #181
Fixes #190 
Fixes #192 
Fixes #194
Fixes #195 
Fixes #197 
Fixes #198 
Fixes #199

### CODE REVIEW

File: knowledge-base/anchore/sbom-action/action-security.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: "Anchore SBOM Action"
```
Comment (h0x0er) [inline knowledge-base/anchore/sbom-action/action-security.yml:1]: Resolves #194

File: knowledge-base/golangci/golangci-lint-action/action-security.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: "Run golangci-lint"
```
Comment (h0x0er) [inline knowledge-base/golangci/golangci-lint-action/action-security.yml:1]: Resolves #197

File: knowledge-base/snyk/actions/golang/action-security.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+name: "Snyk Golang"
```
Comment (h0x0er) [inline knowledge-base/snyk/actions/golang/action-security.yml:1]: Resolves #198

File: knowledge-base/Azure/static-web-apps-deploy/action-security.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: "Azure Static Web Apps Deploy"
+github-token:
+  action-input:
+    input: repo_token # CHECKOUT: https://github.com/Azure/static-web-apps-deploy/blob/v1/action.yml#L31
+    is-default: false
+  permissions:
+    pull-requests: write
```
Comment (h0x0er) [inline knowledge-base/Azure/static-web-apps-deploy/action-security.yml:7]: https://github.com/Azure/static-web-apps-deploy/blob/v1/action.yml#L32

File: knowledge-base/golangci/golangci-lint-action/action-security.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+github-token:
+  action-input: 
+    input: github-token
+    is-default: true
+  permissions:
+    pull-requests: read
+    pull-requests-reason: to fetch pull requests
```
Comment (h0x0er) [inline knowledge-base/golangci/golangci-lint-action/action-security.yml:8]: https://github.com/golangci/golangci-lint-action/blob/ebc78f9b2fcd19a74da6a693bc1419e3313167a8/src/run.ts#L41

File: knowledge-base/anchore/sbom-action/action-security.yml (hunk 1 above)
Comment (varunsh-coder) [reply knowledge-base/anchore/sbom-action/action-security.yml:1]: @h0x0er can you please add link to reason (where it is mentioned that this needs artifacts access), like you did for `static-web-apps-deploy`? You can add it as a comment for `actions-reason`.

File: knowledge-base/docker/metadata-action/action-security.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+name: 'Docker Metadata action'
+github-token:
+  action-input:
+    input: github-token
+    is-default: true
+  permissions:
+    contents: read
+    contents-reason: to query repo information
```
Comment (varunsh-coder) [inline knowledge-base/docker/metadata-action/action-security.yml:8]: Same here. We should start adding link to reason next to the `-reason`. Makes it easy for me to click on the link and understand/ confirm it.

File: knowledge-base/golangci/golangci-lint-action/action-security.yml (hunk 5 above)
Comment (varunsh-coder) [inline knowledge-base/golangci/golangci-lint-action/action-security.yml:8]: same for this - link to reason in comments.

File: knowledge-base/anchore/sbom-action/action-security.yml (hunk 1 above)
Comment (h0x0er) [reply knowledge-base/anchore/sbom-action/action-security.yml:1]: @varunsh-coder thanks for pointing it. In the upcoming PRs I will make sure to add `reference links`.
For this one below are the links:
*   https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L262
 *  https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L459
 * https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L465

Comment (varunsh-coder) [reply knowledge-base/anchore/sbom-action/action-security.yml:1]: @h0x0er upload release asset requires `contents: write` permission
https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L465

File: knowledge-base/anchore/sbom-action/action-security.yml [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+github-token:
+  action-input:
+    input: github-token
+    is-default: true
+  permissions:
+    actions: write
+    actions-reason: to upload & delete workflow artifacts & assets  # CHECKOUT: https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L262
```
Comment (varunsh-coder) [inline knowledge-base/anchore/sbom-action/action-security.yml:8]: @h0x0er I did not see it deleting workflow artifact. You will need to split this into two
1. actions: write for uploading workflow artifact
2. contents: write for uploading release artifact

File: knowledge-base/github/super-linter/slim/action-security.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+name: 'Super-Linter slim'
+github-token:
+  environment-variable-name: GITHUB_TOKEN
+  is-default: false
+  permissions:
+    checks: write
+    checks-reason: to update Checks Section # CHECKOUT: https://github.com/github/super-linter#how-to-use
```
Comment (varunsh-coder) [inline knowledge-base/github/super-linter/slim/action-security.yml:7]: @h0x0er `super-linter` uses statuses as per my earlier analysis. so `super-linter/slim` might be using the same. You can use the same text as well. 
https://github.com/step-security/secure-workflows/blob/1bd3f566c0c735e3039ae1d58d8763eb6a3f9a7b/knowledge-base/github/super-linter/action-security.yml#L6
https://github.com/github/super-linter/blob/feb45bb0350fec6cc687042ed3385154e72d6f89/lib/linter.sh#L644

File: knowledge-base/anchore/sbom-action/action-security.yml (hunk 7 above)
Comment (h0x0er) [reply knowledge-base/anchore/sbom-action/action-security.yml]: @varunsh-coder I have made the required changes.
As per here [here](https://github.com/anchore/sbom-action/blob/3626d7d7b13e87ee6c6f9ded3940dea05a3967bc/src/github/SyftGithubAction.ts#L459), the action is deleting release assets

File: knowledge-base/github/super-linter/slim/action-security.yml (hunk 8 above)
Comment (h0x0er) [reply knowledge-base/github/super-linter/slim/action-security.yml]: @varunsh-coder resolved this issue.

**Pull request merged**